src/Authoring/WinRT.SourceGenerator/AotOptimizer.cs

```diff
@@ -529,10 +530,13 @@ private static List<VtableAttribute> GetVtableAttributesToAddOnLookupTable(Gener
                             {
                                 var argumentClassTypeSymbol = argumentType.Type;
 
-                                // This handles the case where a generic delegate is passed.
+                                // This handles the case where a generic delegate is passed as an object.
+                                // When they are not passed as an object, the marshaler makes sure
+                                // the vtable information is available.
                                 if (argumentClassTypeSymbol.TypeKind == TypeKind.Delegate &&
                                     argumentClassTypeSymbol.MetadataName.Contains("`") &&
-                                    GeneratorHelper.IsWinRTType(argumentClassTypeSymbol))
+                                    GeneratorHelper.IsWinRTType(argumentClassTypeSymbol) &&
+                                    methodSymbol.Parameters[paramsIdx].Type.SpecialType == SpecialType.System_Object)

```

Comment:
Not clear why this additional check is required now. Maybe a comment would help.

Reply:
The comment I had originally changed was intended to do that.  I adjusted the comment to make it clearer.